low-level/solution/lr2/support_bundle/sspl_support_bundle

```diff
@@ -0,0 +1,394 @@
+        self.iem_log_dir = Conf.get(SSPL_CONF, "%s>log_file_path" %
+                                    (self.IEM_SENSOR)).replace("/iem_messages", "")
+
+    def generate(self, args):
+        """Generate support bundle."""
+        logger.info("Starting SSPL bundle generation")
+        self.prepare_bundle(args.bundle_id, args.target_path, args.noencl)
+        self.__clear_tmp_files()
+        status = SupportBundle().get_status(args.bundle_id)
+        print(status)
+
+    @staticmethod
+    def get_status(bundle_id: str):
+        """Get status of support bundle."""
+        status = None
+        req_id = "sb_%s" % bundle_id
+        sspl_sb_id = req_register.get(["%s>sspl_sb_id" % req_id])
+        request_status = BundleRequestHandler.get_processing_status(
+            sspl_sb_id[0])
+        # Extract sspl sb request status from the message.
+        # Example : "Failed - error(22): Invalid request"
+        status = request_status.split(" ")[0]
+        if status != "In-progress":
+            return status
+        return None
+
+    def prepare_bundle(self, bundle_id: str, target_path: str, include_encl):
+        """Collect sspl support bundle data."""
+        bundle_name = "sspl_" + bundle_id
+        sspl_sb_id = self.get_sspl_bundle_data(target_path, include_encl)
+        bundle_location = self._generate_tar(bundle_name)
+        SupportBundle._update_bundle_request_info(
+            bundle_id, sspl_sb_id, bundle_location)
+
+    def get_sspl_bundle_data(self, file_path, include_encl):
+        """Collect SSPL data for support bundle."""
+        request_id = datetime.strftime(datetime.now(), '%Y%m%d%H%M%S%f')
+        BundleRequestHandler._add_bundle_request(request_id, file_path)
+
+        if file_path:
+            if os.path.exists(file_path):
+                file_path = file_path+"/sspl/"
+                self.tmp_bundle_path = os.path.normpath(file_path)
+            else:
+                msg = "Given path %s doesn't exist" % (file_path)
+                self.__clear_tmp_files()
+                raise SupportBundleError(1, msg)
+        os.makedirs(self.tmp_bundle_path, exist_ok=True)
+        try:
+            DEFAULT_STATE = Conf.get(
+                SSPL_CONF, f"{SYSTEM_INFORMATION}>{SSPL_STATE}", "active")
+            try:
+                sspl_state = open(
+                    f"/var/{sspl_const.PRODUCT_FAMILY}/sspl/data/state.txt")
+                sspl_state = sspl_state.readline().rstrip().split("=")
+                # Capturing enclosure logs only when sspl state is active
+                sspl_state = sspl_state[1] if len(
+                    sspl_state) == 2 else DEFAULT_STATE
+                if not sspl_state == 'active':
+                    logger.info("SSPL is in 'degraded' mode,"
+                                "so enclosure data not collected "
+                                "as part of this node support bundle.")
```

Comment:
I think every node has enclosure/pods attached with it. so even if the SSPL is not active in any node, then also we have to capture enclosure data as per my understanding.
@thavanathan @cdeshmukh Please confirm.

Reply:
Done